src/includes/enriching-events/add-attachment/java.mdx

```diff
@@ -0,0 +1,119 @@
+To add an attachment you can either add it to the scope, pass it to any of the `capture` methods, or manipulate the list of attachments in an `EventProcessor` or `beforeSend`.
+
+
+### Passing attachments to capture
+
+You may pass attachments to any of the `capture` methods, e.g. when capturing an Exception:
+
+```java
+import io.sentry.Attachment;
+import io.sentry.Hint;
+import io.sentry.Sentry;
+
+Sentry.captureException(new IllegalStateException(), Hint.withAttachment("/path/to/file.txt"))
+```
+
+```kotlin
+import io.sentry.Attachment;
+import io.sentry.Hint;
+import io.sentry.Sentry;
+
+Sentry.captureException(IllegalStateException(), Hint.withAttachment("/path/to/file.txt"))
+```
+
+
+### Adding attachments in EventProcessor
+
+You may also manipulate attachments in `EventProcessor`:
+
+```java
+import io.sentry.Attachment;
+import io.sentry.EventProcessor;
+import io.sentry.Hint;
+import io.sentry.Sentry;
+import io.sentry.SentryEvent;
+import io.sentry.protocol.Message;
+
+public class MyClass {
+  public void myMethod() {
+    // Add an `EventProcessor` to the current scope. Note that
+    // this helper will process *all* future events in this scope.
+    Sentry.configureScope(scope -> {
+      scope.addEventProcessor(new EventProcessor() {
+        @Override
+        public SentryEvent process(SentryEvent event, Hint hint) {
+          hint.addAttachment(new Attachment("/path/to/file.txt"))
+          return event;
+        }
+      });
+    });
+
+    // Send an event to Sentry. During construction of the event
+    // the attachment will be added by the event processor.
+    Sentry.captureMessage("Hello, world!");
+  }
+}
+```
+
+```kotlin
+import io.sentry.Attachment;
+import io.sentry.EventProcessor;
+import io.sentry.Hint;
+import io.sentry.Sentry
+import io.sentry.SentryEvent;
+import io.sentry.protocol.Message
+
+class MyClass {
+  fun myMethod() {
+    // Add an `EventProcessor` to the current scope. Note that
+    // this helper will process *all* future events in this scope.
+    Sentry.configureScope { scope ->
+      scope.addEventProcessor(object: EventProcessor {
+        override fun process(event: SentryEvent, hint: Hint): SentryEvent? {
+          hint.addAttachment(Attachment("/path/to/file.txt"))
+          return event
+        }
+      })
+    }
+
+    // Send an event to Sentry. During construction of the event
+    // the attachment will be added by the event processor.
+    Sentry.captureMessage("Hello, world!")
+  }
+}
+```
+
+
+### Adding attachments in beforeSend
+
+Another way of adding attachments is using `beforeSend`:
+
+```java
+import io.sentry.Attachment;
+import io.sentry.Hint;
+import io.sentry.Sentry;
+
+Sentry.init(options -> {
```

Comment:
That is gonna be visible for Android too but then its `SentryAndroid`.init`, maybe we also add the `android.mdx` or skip the `Sentry.init` block and focus on the `setBeforeSend` callback declaration.

Reply:
updated